src/dotnet/CarbonAware.CLI/Program.cs

```diff
@@ -1,14 +1,42 @@
-﻿namespace CarbonAwareCLI;
+﻿namespace CarbonAware.CLI;
+
+using CarbonAware.Aggregators.CarbonAware;
+using CarbonAware.Aggregators.Configuration;
+using Microsoft.Extensions.DependencyInjection;
 
 class Program
 {
-    static void Main(string[] args)
-    {
-        var cli = new CarbonAwareCLI(args);
+    private static ICarbonAwareAggregator _aggregator;
+    public static void Main(string[] args)
+    {   
+        InitializePlugin();
+
+        var cli = new CarbonAwareCLI(args, _aggregator);
+
         if (cli.Parsed)
         {
             var emissions = cli.GetEmissions();
-            cli.OutputEmissionsData(emissions);
+            cli.OutputEmissionsData(emissions.Result);
         }
     }
+
+    private static void InitializePlugin() {
```

Comment:
Maybe this should return the `Aggregator`. Also renaming the function to accommodate what it does would be very helpful.